Ecal/src/Ecal/CLUE.cxx

```diff
@@ -34,64 +34,66 @@ T CLUE::dist(T x1, T y1, T z1, T x2, T y2, T z2) {
    likely merged => recluster Did not quite work and I don't remember the idea
    anymore but leaving the code here for inspo */
 
-// void electronSeparation(std::vector<ldmx::EcalHit> hits_) {
-//   std::vector<double> layerThickness =
-//   { 2., 3.5, 5.3, 5.3, 5.3, 5.3, 5.3, 5.3, 5.3, 5.3, 5.3, 10.5, 10.5, 10.5, 10.5,
-//   10.5 }; double air = 10.; std::sort(hits_.begin(), hits_.end(), [](const
-//   ldmx::EcalHit& a, const ldmx::EcalHit& b) {
-//       return a.getZPos() < b.getZPos();
-//   });
-//   std::vector<ldmx::EcalHit> firstLayers;
-//   std::vector<IntermediateEcalCluster> firstLayerClusters;
-//   int layer_index = 0;
-//   double layerZ = hits_[0].getZPos();
-//   for (const auto& hit : hits_) {
-//     if (hit.getZPos() > layerZ + layerThickness[layer_index] + air) {
-//       layer_index++;
-//       // if (layer_index > limit) break;
-//       break;
-//     }
-//     firstLayers.push_back(hit);
-//     firstLayerClusters.push_back(IntermediateEcalCluster(hit, layer_index));
-
-//   }
-//   bool merge = false;
-//   do {
-//     merge = false;
-//     for (int i = 0; i < firstLayerClusters.size(); i++) {
-//       if (firstLayerClusters[i].empty()) continue;
-//       // if (firstLayerClusters[i].centroid().E() >= seed_threshold_) {
-//       for (int j = i + 1; j < firstLayerClusters.size(); j++) {
-//         if (firstLayerClusters[j].empty()) continue;
-//         if (dist(firstLayerClusters[i].centroid().Px(),
-//         firstLayerClusters[i].centroid().Py(),
-//         firstLayerClusters[j].centroid().Px(),
-//         firstLayerClusters[j].centroid().Py()) < 8.) {
-//           firstLayerClusters[i].add(firstLayerClusters[j]);
-//           firstLayerClusters[j].clear();
-//           merge = true;
-//         }
-//       }
-//       // } else break;
-//     }
-//   } while (merge);
-//   ldmx_log(trace) << "--- ELECTRON SEPARATION ---";
-//   for (int i = 0; i < firstLayerClusters.size(); i++) {
-//     if (firstLayerClusters[i].empty()) continue;
-//     ldmx_log(trace) << "  Cluster " << i << " x_: "
-//                     << firstLayerClusters[i].centroid().Px() << " y_: "
-//                     << firstLayerClusters[i].centroid().Py();
-//     for (int j = i + 1;
-//     j < firstLayerClusters.size(); j++) {
-//       if (firstLayerClusters[j].empty()) continue;
-//       auto d = dist(firstLayerClusters[i].centroid().Px(),
-//       firstLayerClusters[i].centroid().Py(),
-//       firstLayerClusters[j].centroid().Px(),
-//       firstLayerClusters[j].centroid().Py());
-//       ldmx_log(trace) << "Dist to cluster " << j << ": " << d;
-//     }
-//   }
-// }
+void CLUE::electronSeparation(std::vector<ldmx::EcalHit> hits) {
+  std::vector<double> layerThickness = {2.,   3.5,  5.3,  5.3, 5.3, 5.3,
+                                        5.3,  5.3,  5.3,  5.3, 5.3, 10.5,
+                                        10.5, 10.5, 10.5, 10.5};
```

Comment:
Worried about this hard-coded geometry stuff especially since its indexed by bi-layer instead of silicon-layer (2 silicon layers per bi-layer).

I think the solution would be to share the `EcalGeometry` condition with `CLUE` but that is technical and I'm willing to just make a new issue and merge this as-is.

Reply:
I agree that's a nicer solution but I really haven't been working on this part. A lot of stuff Ella introduced was touched by Tamas and differences are sure to show up here since I had to patch a lot when merging. 
So yeah in short a separate issue for that sounds great.